quodlibet/qltk/songmodel.py

```diff
@@ -215,18 +217,13 @@ def find(self, song):
                 return iter_
         return
 
-    def find_all(self, songs):
+    def find_all(self, songs: Iterable[Any]):
         """Returns a list of iters for all occurrences of all songs.
         (since a song can be in the model multiple times)
         """
 
         songs = set(songs)
-        found = []
-        append = found.append
-        for iter_, value in self.iterrows():
-            if value in songs:
-                append(iter_)
-        return found
+        return [iter_ for iter_, value in self.iterrows() if value in songs]
```

Comment:
@declension btw, I just compared these two options with the timeit module, since I keep seeing this pattern in the quodlibet codebase. I have to say, my results are inconclusive. None is always better than the other and the difference is usually very small.

Reply:
Cool; personally I think comprehensions are neater but yeah not much in it